Python/compile.c

```diff
@@ -5412,6 +5443,601 @@ compiler_slice(struct compiler *c, expr_ty s)
     return 1;
 }
 
+
+// PEP 634: Structural Pattern Matching
+
+// To keep things simple, all compiler_pattern_* routines follow the convention
+// of preserving TOS (the subject for the given pattern) and pushing either True
+// (match) or False (no match) on top of it. We do this even for irrefutable
+// patterns; the idea is that it's much easier to smooth out any redundant
+// pushing, popping, and jumping in the peephole optimizer than to detect or
+// predict it here.
```

Comment:
NB!

Also, note that the current pattern compiler is designed to be as simple and easy-to-review as possible. The PEP gives us broad freedom to do trickier stuff, but I'm not going to begin work on a rewrite until we have a couple of releases using this simpler approach.